cle/backends/elf/elf.py

```diff
@@ -133,17 +132,22 @@ def __init__(self, binary, addend=None, **kwargs):
         if self.has_dwarf_info and self.loader._load_debug_info:
             # load DWARF information
             try:
-                dwarf = self.reader.get_dwarf_info()
+                dwarf = self._reader.get_dwarf_info()
             except ELFError:
                 l.warning("An exception occurred in pyelftools when loading the DWARF information%s. "
-                          "Marking DWARF as not available for this binary.",
-                          (" on %s" % binary) if isinstance(binary, str) else "",
-                          exc_info=True)
+                          "Marking DWARF as not available for this binary.", self.binary_basename, exc_info=True)
                 dwarf = None
                 self.has_dwarf_info = False
 
             if dwarf and dwarf.has_EH_CFI():
-                self._load_function_hints_from_fde(dwarf)
+                self._load_function_hints_from_fde(dwarf, FunctionHintSource.EH_FRAME)
+
+        if debug_symbols:
+            self.__process_debug_file(debug_symbols)
+        elif self.loader._load_debug_info and self.build_id:
+            debug_filename = '/usr/lib/debug/.build-id/%s/%s.debug' % (self.build_id[:2], self.build_id[2:])
```

Comment:
Is this path pattern specific to Linux or Debian/Ubuntu?

Reply:
https://fedoraproject.org/wiki/RolandMcGrath/BuildID#Find_files_by_build_ID

seems to be standard